src/client/dfs/dfs.c

```diff
@@ -7939,3 +7939,136 @@ dfs_readdir_with_filter(dfs_t *dfs, dfs_obj_t *obj, dfs_pipeline_t *dpipe, daos_
 	D_FREE(buf_keys);
 	return rc;
 }
+
+int
+dfs_cont_set_owner(daos_handle_t coh, d_string_t user, d_string_t group)
+{
+	uid_t                      uid;
+	gid_t                      gid;
+	daos_key_t                 dkey;
+	d_sg_list_t                sgl;
+	d_iov_t                    sg_iovs[4];
+	daos_iod_t                 iod;
+	daos_recx_t                recxs[4];
+	daos_handle_t              oh;
+	int                        i;
+	struct timespec            now;
+	daos_prop_t               *prop;
+	uint32_t                   props[]   = {DAOS_PROP_CO_LAYOUT_TYPE, DAOS_PROP_CO_ROOTS};
+	const int                  num_props = ARRAY_SIZE(props);
+	struct daos_prop_entry    *entry;
+	struct daos_prop_co_roots *roots;
+	int                        rc;
+
+	prop = daos_prop_alloc(num_props);
+	if (prop == NULL)
+		return ENOMEM;
+
+	for (i = 0; i < num_props; i++)
+		prop->dpp_entries[i].dpe_type = props[i];
+	rc = daos_cont_query(coh, NULL, prop, NULL);
+	if (rc) {
+		D_ERROR("daos_cont_query() failed, " DF_RC "\n", DP_RC(rc));
+		D_GOTO(out_prop, rc = daos_der2errno(rc));
+	}
+
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_LAYOUT_TYPE);
+	if (entry == NULL || entry->dpe_val != DAOS_PROP_CO_LAYOUT_POSIX) {
+		rc = EINVAL;
+		D_ERROR("container is not of type POSIX: %d (%s)\n", rc, strerror(rc));
+		D_GOTO(out_prop, rc);
+	}
+
+	rc = daos_cont_set_owner(coh, user, group, NULL);
```

Comment:
daos_cont_set_owner requires user or group. Should dfs_cont_set_owner check and report an error above? Or just rely on this call failing?

Reply:
it would be repetitive to add the checking here. daos_cont_set_owner() should properly log an error in this case.